From: Matthew Garrett <matthew.garrett@nebula.com>
Subject: [PATCH V3 09/11] uswsusp: Disable when module loading is restricted
Date: Tue,  3 Sep 2013 19:50:16 -0400

Git-commit: Not yet
Patch-mainline: Not yet, fedora 20
References: fate#314486, bnc#884333
Target: SLE-11 SP3

uswsusp allows a user process to dump and then restore kernel state, which
makes it possible to avoid module loading restrictions. Prevent this when
any restrictions have been imposed on loading modules.

Signed-off-by: Matthew Garrett <matthew.garrett@nebula.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 kernel/power/user.c | 4 ++++
 1 file changed, 4 insertions(+)

Index: linux-3.0-SLE11-SP3/kernel/power/user.c
===================================================================
--- linux-3.0-SLE11-SP3.orig/kernel/power/user.c
+++ linux-3.0-SLE11-SP3/kernel/power/user.c
@@ -23,6 +23,7 @@
 #include <linux/console.h>
 #include <linux/cpu.h>
 #include <linux/freezer.h>
+#include <linux/module.h>
 
 #include <asm/uaccess.h>
 
@@ -69,6 +70,9 @@ static int snapshot_open(struct inode *i
 	struct snapshot_data *data;
 	int error;
 
+	if (secure_modules())
+		return -EPERM;
+
 	mutex_lock(&pm_mutex);
 
 	if (!atomic_add_unless(&snapshot_device_available, -1, 0)) {
